src/com/reason/ide/files/EsyPackageJsonFile.java

```diff
@@ -0,0 +1,59 @@
+package com.reason.ide.files;
+
+import com.intellij.framework.detection.FileContentPattern;
+import com.intellij.json.JsonFileType;
+import com.intellij.json.psi.JsonFile;
+import com.intellij.json.psi.JsonObject;
+import com.intellij.json.psi.JsonValue;
+import com.intellij.lang.Language;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.patterns.ElementPattern;
+import com.intellij.patterns.PatternCondition;
+import com.intellij.psi.FileViewProvider;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.ProcessingContext;
+import com.intellij.util.indexing.FileContent;
+import com.intellij.util.indexing.FileContentImpl;
+import org.jetbrains.annotations.NotNull;
+
+public class EsyPackageJsonFile extends FileBase {
```

Comment:
FileBase is a (IntelliJ)PsiFile that implements a PsiModule interface, aka OCaml or ReasonML source file. 
An Esy file is a json file, it shouldn't extends the FileBase

Reply:
is the PsiElement base type needed to have this icon be provided by IconProvider?

Reply:
ok, so waiting for the plugin.xml modification before merging the PR.

Reply:
did you try it ? how does it work if you have 2 different json file type (esy/package.json and bsconfig.json) ? they have both .json extension.

Reply:
So I am unable to differentiate between package.json types for esy.

However, I will add an entry to target _esy and esy.lock files using the `filenames` attribute which matches filenames and takes precedence over `extensions`.

If you look at my bucklescript icon PR you will see what I mean.

Reply:
Updated. Added entries for `_esy` and `esy.lock` files